woocommerce/google-listings-and-ads
# WooCommerce Subscriptions compatibility: Fix the visible issue of the product data tab and meta box for some unsupported product types

### Changes proposed in this Pull Request:

Closes #2086

- Remove the controls' show/hide CSS classes of the product editing page from PHP side.
- Control the show or hide of the product data tab and meta box added to the product editing page by JS.
- Merge the custom-inputs script into the product-attributes script.

### Screenshots:

https://github.com/woocommerce/google-listings-and-ads/assets/17420811/f8ef39ee-48ec-4161-b5b7-1dc8a2205794

### Detailed test instructions:

1. Install and activate WooCommerce Subscriptions and WooCommerce Product Bundles.
2. Restart the client build script (`npm start`) if it's running.
3. Go to edit a product.
4. Select each product type to see if the "Google Listings and Ads" tab and "Channel visibility" meta box are visible for supported product types and hidden for unsupported ones.
   - Supported product types:
      https://github.com/woocommerce/google-listings-and-ads/blob/f325aa1cebee24ffd95985c3dd719233e045e44a/js/src/product-attributes/index.js#L11-L18
5. Create a product with the "Simple subscription" type and go to its editing page.
6. After the page is initialized, check if the UIs mentioned in step 3 are hidden.

### Changelog entry

> Fix - WooCommerce Subscriptions compatibility: Fix the visible issue of the "Google Listings and Ads" tab and "Channel visibility" meta box for some unsupported product types.
